src/pages/PlaylistDetails/PlaylistDetails.js

```diff
@@ -0,0 +1,160 @@
+  const [isEditMode, setIsEditMode] = useState(false);
+  const [playlistName, setPlaylistName] = useState(playlist?.name);
+  const playlistNameEl = useRef(null);
+
+  const editBtnClickHandler = async () => {
+    if (!isEditMode) {
+      playlistNameEl.current.focus();
+    } else {
+      try {
+        const response = await axios.post(
+          `https://watch-finsight.desaihetav.repl.co/playlists/update/${playlistId}`,
+          {
+            newName: playlistName,
+          }
+        );
+        if (response.data.success) {
+          dispatch({
+            type: "UPDATE_PLAYLIST_NAME",
+            payload: { _id: playlist._id, name: playlistName },
+          });
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    }
+    setIsEditMode((isEditMode) => !isEditMode);
+  };
+
+  return playlist ? (
+    <div className={`container relative ${styles.pageContainer}`}>
+      {showDeleteModal && (
+        <DeleteModal
+          setshowDeleteModal={setshowDeleteModal}
+          dispatch={dispatch}
+          playlistId={playlistId}
+          playlistName={playlistName}
+        />
+      )}
+      <div className="space-y-1"></div>
+      <div className="space-y-1"></div>
+      <div className={`flex justify-between items-center`}>
+        <input
+          readOnly={!isEditMode}
+          ref={playlistNameEl}
+          className={`${styles.name}`}
+          value={playlistName}
+          onChange={(e) => setPlaylistName(() => e.target.value)}
+        />
+
+        <div className={`flex`}>
+          <button
+            onClick={() => editBtnClickHandler()}
+            className={`btn btn-icon btn-small ${
+              isEditMode ? "btn-solid" : "btn-ghost"
+            }`}
+          >
+            <span className="material-icons-outlined">
+              {isEditMode ? "done" : "edit"}
+            </span>
+          </button>
+          <button
+            onClick={() => setshowDeleteModal(() => true)}
+            className="btn btn-icon btn-small btn-ghost"
+          >
+            <span className="material-icons-outlined">delete</span>
+          </button>
+        </div>
+      </div>
+      <ul className="flex flex-col w-full">
+        {playlist?.videos.length !== 0 ? (
+          playlist?.videos?.map((videoItem) => (
+            <li key={videoItem} className="w-full">
+              <VideoCard videoItemId={videoItem} />
+            </li>
+          ))
+        ) : (
+          <div className="flex flex-col items-center justify-center flex-grow-1">
+            <h2 className={`w-full text-center my-8`}>
+              No video is this playlist
+            </h2>
+            <Link to="/" className={`btn btn-solid`}>
+              Explore
+            </Link>
+          </div>
+        )}
+      </ul>
+    </div>
+  ) : (
+    <Loading />
+  );
+}
+
+const DeleteModal = ({ setshowDeleteModal, playlistId, playlistName }) => {
```

Comment:
Delete Modal can be extracted out as a Component here.
Also, if possible, try to use one modal with different variants passed as props, to change theor functionality accordingly and thus reduce code.